yb-voyager/cmd/exportData.go

```diff
@@ -798,6 +807,200 @@ func getInitialTableList() (map[string]string, []sqlname.NameTuple) {
 	return partitionsToRootTableMap, finalTableList
 }
 
+func getInitialTableList() (map[string]string, []sqlname.NameTuple) {
+	// store table list after filtering unsupported or unnecessary tables
+	msr, err := metaDB.GetMigrationStatusRecord()
+	if err != nil {
+		utils.ErrExit("error fetching migration status record: %v", err)
+	}
+
+	if startClean || len(msr.TableListExportedFromSource) == 0 {
+		// fresh start case or the first run where we don't have a table list stored in msr
+		return fetchTablesNamesFromSourceAndFilterTableList()
+
+	}
+
+	//resumption case where we will use a table-list stored in
+	var fullTableList []sqlname.NameTuple
+	var finalTableList []sqlname.NameTuple
+	storedTableList := msr.TableListExportedFromSource
+	if source.DBType == POSTGRESQL && msr.SourceExportedTableListWithLeafPartitions != nil {
+		//list with partitioned root table and all leafs
+		storedTableList = msr.SourceExportedTableListWithLeafPartitions
+	}
+	for _, t := range storedTableList {
+		var tuple sqlname.NameTuple
+		var err error
+		switch source.DBType {
+		case POSTGRESQL:
+			defaultSchemaName, _ := getDefaultSourceSchemaName()
+			obj := sqlname.NewObjectNameWithQualifiedName(source.DBType, defaultSchemaName, t)
+			//get the name tupe via `getNAmeFromObj` function in case its leaf partition
+			tuple = getNameTupleFromObjectName(obj)
+		default:
+			tuple, err = namereg.NameReg.LookupTableName(t)
+			if err != nil {
+				utils.ErrExit("lookup for table name failed: %s: %v", t, err)
+			}
+		}
+		fullTableList = append(fullTableList, tuple)
+	}
+	finalTableList = fullTableList
+
+	var partitionsToRootTableMap map[string]string
+	switch source.DBType {
```

Comment:
Is this entire block only to support upgrade scenario? Let's discuss this. I don't think we should put extra effort into supporting upgrade scenario for this new feature. We should just ensure that upgrade doesn't fail.